### PR TITLE
Create modal in CU030

### PR DESCRIPTION
**Pull request by novadeveloper14**
The modal that appears in operation CU030 was carried out, which is when the reject button is clicked in the financialReporting outlet.

image

points to consider:

- The modal component was configured to be general and accept any type of configuration related to the textarea

- The textarea component of the current version of the design-system library does not allow removing the resize

- The component receives a maxLength, which is the characters maximum that the component receives, which cannot be removed, so the modal was assigned to have this value be 200 characters by default.

- when the number of current characters in the textarea is equal to 0 or is greater than the maxLength the confirm button becomes disabled

### CODE REVIEW

**Review by LinconOssa — APPROVED** [review]
Meets the requirements.

File: src/components/layout/ContainerSections/index.tsx [hunk 1]
```diff
@@ -11,40 +14,63 @@ interface IContainerSectionsProps {
 
 export const ContainerSections = (props: IContainerSectionsProps) => {
   const { children } = props;
+  const [showRejectModal, setShowRejectModal] = useState(false);
```
Comment (carin91) [inline src/components/layout/ContainerSections/index.tsx:17]: Change the spelling of the word.
Reject = Rechazar
Rejection = Rechazo

File: src/components/modals/TextAreaModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import {
```
Comment (carin91) [inline src/components/modals/TextAreaModal/index.tsx:1]: Every component must have its storybook. Implement the storybook for this component.

File: src/components/modals/TextAreaModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  const node = document.getElementById(portalId);
+
+  if (node === null) {
+    throw new Error(
+      "The portal node is not defined. This can occur when the specific node used to render the portal has not been defined correctly."
+    );
+  }
+
+  return createPortal(
+    <Blanket>
```
Comment (carin91) [inline src/components/modals/TextAreaModal/index.tsx:72]: Is it a form or not? it has inputs (type="textArea") as the submit button. Why avoid the form tag of html? The submit will also trigger a post request, therefore, I would use the form tag.

Currently, we are using Formik in several apps (such as PortalClientesFondecom and iSetting Portal), so, I would use Formik too.

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/components/modals/TextAreaModal/index.tsx (hunk 2 above)
Comment (carin91) [reply src/components/modals/TextAreaModal/index.tsx:1]: This task is still pending.

(1) The storybook is not working. To adjust that you need to create a router as decoration of the storybook.
(2) Change the location of the storybook.
image

  title: "components/layouts/ContainerSections",

**Review by carin91 — APPROVED** [review]
The component has a little difference with the wireframes, verify that this component is implemented as required.

image

(2) I understand that this is a mistake of the designer, however, avoid errors in spelling and gramatics.

Error: "Motivo del Rechazo"
Correct: "Motivo del rechazo."

**Pull request merged**